editor/project_manager.cpp

```diff
@@ -1838,16 +1840,22 @@ ProjectManager::ProjectManager() {
 	Label *sort_label = memnew(Label);
 	sort_label->set_text(TTR("Sort:"));
 	sort_filters->add_child(sort_label);
-	Vector<String> vec1;
-	vec1.push_back("Name");
-	vec1.push_back("Last Modified");
+	Vector<String> sort_filter_titles;
+	sort_filter_titles.push_back("Name");
+	sort_filter_titles.push_back("Last Modified");
 	project_order_filter = memnew(ProjectListFilter);
-	project_order_filter->_setup_filters(vec1);
+	project_order_filter->_setup_filters(sort_filter_titles);
 	project_order_filter->set_filter_size(150);
 	sort_filters->add_child(project_order_filter);
 	project_order_filter->connect("filter_changed", this, "_load_recent_projects");
 	project_order_filter->set_custom_minimum_size(Size2(180, 10) * EDSCALE);
 
+	Variant projects_sorting_order = EditorSettings::get_singleton()->get("project_manager/sorting_order");
+
+	if (projects_sorting_order.get_type() != Variant::NIL) {
+		project_order_filter->set_filter_option((ProjectListFilter::FilterOption)((int)projects_sorting_order));
```

Comment:
You might be able to do this without the (int) cast if you add `VARIANT_ENUM_CAST`.

Reply:
Well, given that this cast is only needed in this one place, leaving it like this might be simplier. And I'll need to make the enum public to use VARIANT_ENUM_CAST, I guess. (not sure why it's private now, but still :)).

Reply:
I'd do `int project_sorting_order = (int)EditorSettings::get_singleton()->get("project_manager/sorting_order");` to avoid this `Variant::NIL` stuff.